test/Sema/diag_erroneous_iuo.swift

```diff
@@ -116,7 +116,7 @@ _ = [Int!]() // expected-error {{'!' is not allowed here; perhaps '?' was intend
 let _: [Int!] = [1] // expected-error {{'!' is not allowed here; perhaps '?' was intended?}}{{12-13=?}}
 _ = Optional<Int!>(nil) // expected-error {{'!' is not allowed here; perhaps '?' was intended?}}{{17-18=?}}
 let _: Optional<Int!> = nil // expected-error {{'!' is not allowed here; perhaps '?' was intended?}}{{20-21=?}}
-_ = Int!?(0) // expected-error {{'!' is not allowed here; perhaps '?' was intended?}}{{8-9=?}}
+_ = Int!?(0) // expected-error 3 {{'!' is not allowed here; perhaps '?' was intended?}}{{8-9=?}}
```

Comment:
@xedin I don't know how concerned we are about this case, but it's triple-diagnosing because precheck and the solver are resolving the underlying type separately. The only thing stopping it from doing that before was that it was storing the instance type in the TypeLoc and using that to bypass type resolution.

Reply:
That's unfortunate but as far as I understand if we stop trying to lookup members in `resolveDeclRefExpr` this should no longer be a problem, is that correct?

Reply:
I bet that tackles one of them. The one in precheck was always suspicious because it relied on the order of validation. That leaves the only strictly necessary one: The one in the solver.